ouster_client/CMakeLists.txt

```diff
@@ -20,18 +20,39 @@ add_custom_command(TARGET generate_build_header PRE_BUILD
   -P ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/VersionGen.cmake)
 
 add_library(ouster_build INTERFACE)
-target_include_directories(ouster_build INTERFACE ${CMAKE_CURRENT_BINARY_DIR}/include)
+target_include_directories(ouster_build INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>)
 add_dependencies(ouster_build generate_build_header)
 
 # ==== Libraries ====
 add_library(ouster_client src/client.cpp src/types.cpp src/netcompat.cpp src/lidar_scan.cpp src/image_processing.cpp)
-target_link_libraries(ouster_client PUBLIC jsoncpp_lib Eigen3::Eigen ouster_build)
+target_link_libraries(ouster_client
+  PUBLIC Eigen3::Eigen $<BUILD_INTERFACE:ouster_build>
+  PRIVATE jsoncpp_lib)
+target_compile_definitions(ouster_client PUBLIC EIGEN_MAX_ALIGN_BYTES=32 EIGEN_MPL2_ONLY)
```

Comment:
I'm wondering, is this part of #267? We are having troubles with ``EIGEN_MAX_ALIGN_BYTES=32`` on our embedded aarch64 system with custom buildroot build of PCL and ROS and Eigen 3.3.7. We get segfaults of this type:

```
==8092== Invalid read of size 8
==8092==    at 0x4880570: handmade_aligned_free (Memory.h:98)
==8092==    by 0x4880570: aligned_free (Memory.h:179)
==8092==    by 0x4880570: conditional_aligned_free<true> (Memory.h:230)
==8092==    by 0x4880570: conditional_aligned_delete_auto<double, true> (Memory.h:416)
==8092==    by 0x4880570: ~DenseStorage (DenseStorage.h:542)
==8092==    by 0x4880570: ~PlainObjectBase (PlainObjectBase.h:98)
==8092==    by 0x4880570: ~Array (Array.h:45)
==8092==    by 0x4880570: ouster_ros::scan_to_cloud(ouster::XYZLut const&, std::chrono::duration<long, std::ratio<1l, 1000000000l> >, ouster::LidarScan const&, pcl::PointCloud<ouster_ros::Point>&) (ros.cpp:70)
==8092==    by 0x11931F: operator() (os_cloud_node.cpp:141)
==8092==    by 0x11931F: boost::detail::function::void_function_obj_invoker1<main::{lambda(ouster_ros::PacketMsg_<std::allocator<void> > const&)#1}, void, ouster_ros::PacketMsg_<std::allocator<void> > const&>::invoke(boost::detail::function::function_buffer&, ouster_ros::PacketMsg_<std::allocator<void> > const&) (function_template.hpp:158)
==8092==    by 0x11DE23: operator() (function_template.hpp:763)
==8092==    by 0x11DE23: ros::SubscriptionCallbackHelperT<ouster_ros::PacketMsg_<std::allocator<void> > const&, void>::call(ros::SubscriptionCallbackHelperCallParams&) (subscription_callback_helper.h:144)
==8092==    by 0x4F8F53B: ros::SubscriptionQueue::call() (in /opt/ros/lib/libroscpp.so)
==8092==    by 0x4F611EF: ros::CallbackQueue::callOneCB(ros::CallbackQueue::TLS*) (in /opt/ros/lib/libroscpp.so)
==8092==    by 0x4F61863: ros::CallbackQueue::callAvailable(ros::WallDuration) (in /opt/ros/lib/libroscpp.so)
==8092==    by 0x4F9075F: ros::SingleThreadedSpinner::spin(ros::CallbackQueue*) (in /opt/ros/lib/libroscpp.so)
==8092==    by 0x4F8092F: ros::spin() (in /opt/ros/lib/libroscpp.so)
==8092==    by 0x11706B: main (os_cloud_node.cpp:173)
==8092==  Address 0x948ecb8 is 8 bytes before a block of size 786,432 alloc'd
==8092==    at 0x4844004: malloc (in /usr/lib/valgrind/vgpreload_memcheck-arm64-linux.so)
==8092== 
==8092== Invalid free() / delete / delete[] / realloc()
==8092==    at 0x4845088: free (in /usr/lib/valgrind/vgpreload_memcheck-arm64-linux.so)
==8092==  Address 0x29000000000 is not stack'd, malloc'd or (recently) free'd
==8092==
```

When this definition is removed the problem disappears.

Our setup is quite complicated so it is hard to describe precisely all the options, but I just wanted to understand why you added this definition here (performance, ABI compatibility, etc.) and whether it is critical for some functionality. I couldn't find any detailed description that could be relevant except what's already in #267.

Thank you.

Reply:
I believe this was fixed in #313